protocols/identify/src/identify.rs

```diff
@@ -254,6 +272,16 @@ impl NetworkBehaviour for Identify {
     ) {
         match event {
             IdentifyHandlerEvent::Identified(info) => {
```

Comment:
While we do have a limit of `4096 bytes` per identify message, what do you think of enforcing a limit on the number of addresses here anyways?

https://github.com/libp2p/rust-libp2p/blob/937b59de8946eb3773e292691dac4e5f3e081706/protocols/identify/src/protocol.rs#L204-L206

Reply:
What do you think would be an appropriate limit?

Unless we can inform the other peer somehow, silently capping the number of addresses here could result in a nasty bug.

I think the overall message size is a good enough limit that will wake someone somewhere when it is hit! (Assuming the connection gets closed if the message is too large)